src/main/scala/io/iohk/ethereum/mpt/Node.scala

```diff
@@ -1,63 +1,71 @@
 package io.iohk.ethereum.mpt
 
-import io.iohk.ethereum.mpt.MerklePatriciaTrie.HashFn
+import akka.util.ByteString
+import io.iohk.ethereum.crypto
+import io.iohk.ethereum.domain.Account
+import io.iohk.ethereum.network.p2p.messages.PV63.AccountImplicits
 import io.iohk.ethereum.rlp.{encode => encodeRLP}
 
 /**
   * Trie elements
   */
-private[mpt] sealed trait Node {
+sealed trait MptNode {
 
   import MerklePatriciaTrie._
 
-  protected val hashFn: HashFn
+  lazy val encode: Array[Byte] = encodeRLP[MptNode](this)
 
-  lazy val encode: Array[Byte] = encodeRLP[Node](this)
+  lazy val hash: Array[Byte] = Node.hashFn(encode)
 
-  lazy val hash: Array[Byte] = hashFn(encode)
-
-  def capped: Array[Byte] = {
+  def capped: ByteString = {
     val encoded = encode
-    if (encoded.length < 32) encoded else hash
+    if (encoded.length < 32) ByteString(encoded) else ByteString(hash)
   }
 }
 
-private[mpt] case class LeafNode(key: Array[Byte], value: Array[Byte], hashFn: HashFn) extends Node
+object Node {
+  val hashFn: (Array[Byte]) => Array[Byte] = (input: Array[Byte]) => crypto.kec256(input)
+}
+
+case class LeafNode(key: ByteString, value: ByteString) extends MptNode {
+  import AccountImplicits._
+  def getAccount: Account = value.toArray[Byte].toAccount
```

Comment:
As the MPT is used for other things apart from accounts, maybe we should move this function somewhere else?